### ISSUE TITLE
Perhaps WaitFor... default timeout should not be waiting indefinitely?

### ISSUE THREAD

**Issue by jetersen** [mariotoffia/FluentDocker#228]
WaitForHealthy uses `Timespan timeout = default` which is indefinitely.
WaitForPort uses `double millisTimeout = double.MaxValue` which also feels like indefinitely.

Perhaps it would be better to have a 10 minute default timeout as in CI/CD or Test scenario you could be waiting for a container to never recover.
This could turn into a costly endeavor when your paying minutes for build time (GitHub Actions 😉)
User can always overwrite the timeout periode if they need more time.

**Comment by mariotoffia** [comment]
Hmm... or maybe have a setting that you can control e.g. environment variable FLUENTDOCKER_WAITFOR_TIMEOUT that you can set as a DevOps?

**Comment by jetersen** [comment]
`FLUENTDOCKER_WAITFOR_TIMEOUT` could be one option, though I think the better option might be to set a reasonable timeout by default and perhaps check if `FLUENTDOCKER_WAITFOR_TIMEOUT` to override the sane default.

**Comment by mariotoffia** [comment]
That's true! 😊 but what is a sane timeout? - may have different timeouts on their specific workloads and environments. 

What I mean by that is that quite a few expect it to run for at least a certain amount of time - changing it now in a non-breaking release number would absolutely break some functionality (I know for certain some inhouse apps would probably break if we set it below 5 - 30 minutes....).

That's why I thought `FLUENTDOCKER_WAITFOR_TIMEOUT` would be an unintrusive way and do not break backwards compatibility but still, you get your ability to provide the timeout externally.

I'm open to suggestions.

Cheers,
 Mario 😊

**Comment by mariotoffia** [comment]
**Resolved in v3.0.0.** The v3 ground-up rewrite implements (or fixes) what this issue asked for.

The issue reported that v2's WaitForHealthy used `TimeSpan timeout = default` (indefinite) and WaitForPort used `double millisTimeout = double.MaxValue` (effectively indefinite), which is costly in CI scenarios where a container never recovers. The requested fix was a sane finite default (the maintainer suggested ~10 minutes or an env-var override). In v3 every wait builder method now has a finite default timeout of 30000 ms (30 seconds), and the underlying service-level wait extensions default to the same 30000 ms — there is no remaining indefinite-default path. A repo-wide search for indefinite-timeout sentinels (double.MaxValue / long.MaxValue / `= default` near wait/timeout) in the wait code returned nothing relevant (the single Timeout.Infinite hit is an unrelated cancellation-race idiom). v3 also added a clear fast-fail when WaitForHealthy is used on an image with no HEALTHCHECK,…

<sub>Where this lives in v3: FluentDocker/Builders/IContainerBuilder.cs: WaitForPort(string, long timeoutMs = 30000) [lines 285, 294], WaitForProcess(..., long timeoutMs = 30000) [302], WaitForHttp(..., long timeoutMs = 30000) [311, 329], WaitForLogMessage(..., long timeoutMs = 30000) [341], WaitForHealthy(long timeoutMs = 30000) [348] — all finite defaults. FluentDocker/Services/Extensions/ServiceExtensions.cs: WaitForPortAsync (long timeout = 30000) [98-159], WaitForProcessAsync (=30000) [169], WaitForHttpAsync (=30000) [208], WaitForLogMessageAsync (=30000) [256] and the sync wrappers [334-361] all default to 30000ms. FluentDocker/Builders/ContainerBuilder.WaitConditions.cs: WaitForHealthyAsync [206-238] loops…</sub>

Closing as part of the v3 issue sweep. If your scenario still isn't covered by v3, please reopen with a v3 reproduction and we'll take another look. Thanks for the report! 🙏
